### ISSUE TITLE
About the Database.asynchronous() method

### ISSUE THREAD

**Issue by yuchen99** [davidmoten/rxjava-jdbc#16]
Hi gays：
I user rxNetty + rxjava-jdbc for a project,when I  use the asynchronous api for ab test
like `ab -k  -n 100000 -c 1000 'http://localhost:8080/aa?user_id=4' `.
It will create so many threads  at last throw  OutOfMemoryError
```
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded
```
 *.I din't know why,it's something in my code ?
On other hand 
Is that create some new api for set an new Scheduler by user ,make user can control what kind Scheduler to use 
some thing 
```
 Database.asynchronous(Func0<Scheduler> asynScheduler) 
```
here is problem  code
```java
public Observable<User> findUser(int id) {
    return db.asynchronous().select("select * from users where id = ?").parameter(id).autoMap(User.class);
  }
```
In rxNetty server 
```java
RxNetty.newHttpServerBuilder(8080, new RequestHandler<ByteBuf, ByteBuf>() {
      @Override
      public Observable<Void> handle(HttpServerRequest<ByteBuf> request, final HttpServerResponse<ByteBuf> response) {
        request.getQueryParameters().get("user_id").get(0);
       int user_id= Ints.tryParse(Iterables.getFirst(request.getQueryParameters().get("user_id"),"1"));
        return us.findUser(user_id)
          .map(new Func1<User, String>() {
            @Override
            public String call(User user) {
              return JSON.toJSONString(user);
            }
          })
          .flatMap(new Func1<String, Observable<Void>>() {
            @Override
            public Observable<Void> call(String s) {
              response.getHeaders().add(HttpHeaders.Names.CONTENT_LENGTH, s.getBytes().length);
              response.writeStringAndFlush(s);
//              System.out.printf("in flat map thread [%s] \n",Thread.currentThread().getName());
              return response.close();
            }
          });
      }
    }).eventLoop(new EpollEventLoopGroup()).channel(EpollServerSocketChannel.class).build().startAndWait();
```

**Comment by yuchen99** [comment]
Sorry In `Constructor` method can do that

**Comment by davidmoten** [comment]
Hi, is your problem resolved? I didn't understand your last message.

**Comment by yuchen99** [comment]
Yes.
When I remove the `asynchronous` method it work well, because it use the current thread.
But when I use `asynchronous`, it use other threads to handle db queries, when there are a lot of concurrent connections, it will create so many threads, finally OOM 
So I suggest to provide a method, let the user decides to use different scheduler

**Comment by davidmoten** [comment]
That's a good suggestion. The scheduler used is ```Schedulers.io()``` which is unbounded. I'll adjust the api to allow the user to specify a scheduler.

**Comment by davidmoten** [comment]
By the way here is a workaround for you. You can use the Database constructor as below to get what you want:

```java
ConnectionProvider cp = new ConnectionProviderPooled(url, minPoolSize, maxPoolSize);
Database db = new Database(cp, () -> Schedulers.computation());
```

**Comment by davidmoten** [comment]
I'll deploy a new version to Maven Central in about 3 hours.

**Comment by yuchen99** [comment]
Awesome!!!

**Comment by davidmoten** [comment]
Released to Maven Central 0.5.6
